### ISSUE TITLE
Add compatibility with gradle configuration cache

### ISSUE THREAD

**Issue by masmc05** [Minecrell/plugin-yml#32]
When trying to use this plugin with configuration cache we have these issues, would be glad if you'll be able to add this compatibility
```
FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':script-host:generateBukkitPluginDescription'.
> Cannot fingerprint input property 'librariesConfiguration': value 'configuration ':script-host:bukkitLibrary'' cannot be serialized.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
Configuration cache problems found in this build.

3 problems were found storing the configuration cache, 2 of which seem unique.
- Task `:script-host:generateBukkitPluginDescription` of type `net.minecrell.pluginyml.GeneratePluginDescription`: cannot serialize object of type 'org.gradle.api.internal.artifacts.configurations.DefaultConfiguration', a subtype of 'org.gradle.api.artifacts.Configuration', as these are not supported with the configuration cache.
  See https://docs.gradle.org/8.0.2/userguide/configuration_cache.html#config_cache:requirements:disallowed_types
- Task `:script-host:generateBukkitPluginDescription` of type `net.minecrell.pluginyml.GeneratePluginDescription`: cannot serialize object of type 'org.gradle.api.internal.project.DefaultProject', a subtype of 'org.gradle.api.Project', as these are not supported with the configuration cache.
  See https://docs.gradle.org/8.0.2/userguide/configuration_cache.html#config_cache:requirements:disallowed_types
  
  
See the complete report at [generated file](https://github.com/Minecrell/plugin-yml/files/11211022/configuration-cache-report.zip)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
Configuration cache entry discarded with 3 problems.
```

**Comment by masmc05** [comment]
to enable configuration cache, you may want to add 
```properties
org.gradle.configuration-cache=true
```
to gradle.properties (using gradle version 8.1)

**Comment by stephan-gh** [comment]
Looks like this was fixed already in plugin-yml 0.5.3, please use the latest version. I was only able to find a slight problem with Nukkit that I fixed in https://github.com/Minecrell/plugin-yml/commit/352edde165eb53a6e096bc64aeb444b61c5d835f.
